### PR TITLE
added diag plugin

### PR DESCRIPTION
**Pull request by acaldas**
https://github.com/ripe-tech/ripe-core/issues/3871

### CODE REVIEW

**Comment by hugo-gomes** [comment]
don't forget to update the `README.md`

File: src/js/base/api.js [hunk 1]
```diff
@@ -165,6 +169,9 @@ ripe.Ripe.prototype._requestURL = function(url, options, callback) {
     if (contentType) {
         request.setRequestHeader("Content-Type", contentType);
     }
+
+    this.trigger("pre_request", request, options);
```
Comment (joamag) [inline src/js/base/api.js]: maybe it's better to name this event `build_request` and then have a pre_request on `loadstart`

File: src/js/plugins/base.js [hunk 2]
```diff
@@ -11,7 +11,8 @@ ripe.Ripe.plugins.Plugin = function() {
     ripe.Observable.call(this);
 };
 
-ripe.assign(ripe.Ripe.plugins.Plugin.prototype, ripe.Observable.prototype);
+ripe.Ripe.plugins.Plugin.prototype = Object.create(ripe.Observable.prototype);
```
Comment (joamag) [inline src/js/plugins/base.js]: why this change ? Bug fix ?

File: src/python/ripe_demo/static/js/main.js [hunk 3]
```diff
@@ -7,7 +7,7 @@ var FACES = ["side", "top", "front"];
 window.onload = function() {
     var element = document.getElementById("configurator");
     var _body = document.querySelector("body");
-    var url = _body.dataset.url || "https://sandbox.platforme.com/api/";
+    var url = _body.dataset.url || "http://localhost:8181/api/";  // || "https://sandbox.platforme.com/api/";
```
Comment (joamag) [inline src/python/ripe_demo/static/js/main.js]: :)

File: gulpfile.js [hunk 4]
```diff
@@ -23,13 +23,13 @@ var paths = {
 gulp.task("build-js", () => {
     return gulp
         .src(paths.scripts)
+        .pipe(replace("__VERSION__", _package.version))
```
Comment (joamag) [inline gulpfile.js]: isn't this going to change the source (non packed) files also ?

Comment (acaldas) [reply gulpfile.js]: The changes occur on files on the `dist` folder

File: src/js/plugins/base.js (hunk 2 above)
Comment (acaldas) [reply src/js/plugins/base.js]: It's the same approach used on the plugins
